go/tools/builders/generate_test_main.go

```diff
@@ -235,7 +237,16 @@ func main() {
 		}
 	}
 	{{end}}
-	bzltestutil.RegisterTimeoutHandler()
+
+	testTimeout := os.Getenv("TEST_TIMEOUT")
+	if testTimeout != "" {
+		flag.Lookup("test.timeout").Value.Set(testTimeout+"s")
+		// If Bazel sends a SIGTERM because the test timed out, it sends it to all child processes. Because
+		// we already set -test.timeout according to the TEST_TIMEOUT, we need to ignore the signal so the
+		// test has time to clean up. It will be killed by Bazel after the grace period instead.
+		signal.Ignore(syscall.SIGTERM)
```

Comment:
Could you also update this comment?
https://github.com/bazelbuild/rules_go/blob/e6cb73792b77e3b595ba456e92211ae7e4a82346/go/tools/bzltestutil/wrap.go#L128-L132